lib/spack/spack/patch.py

```diff
@@ -110,7 +111,26 @@ class FilePatch(Patch):
     def __init__(self, pkg, relative_path, level, working_dir,
                  ordering_key=None):
         self.relative_path = relative_path
-        abs_path = os.path.join(pkg.package_dir, self.relative_path)
+
```

Comment:
This seems more appropriate as a function outside the context of the constructor.

Reply:
Creating a separate function would also make it easier to add a test case for lines 130-133.

Reply:
I figured out a way to test the constructor, but whether the logic goes here or elsewhere isn't really relevant to that.

Reply:
The test was the most important issue for me.